File: app/dto/project.py
from __future__ import annotations
from dataclasses import dataclass, field
from typing import Dict, Any, Optional
from core.models import Project

@dataclass
class ProjectSummaryDTO:
    id: str
    name: str
    status: str
    task_count: int

    def to_dict(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "name": self.name,
            "status": self.status,
            "task_count": self.task_count
        }

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> ProjectSummaryDTO:
        return cls(
            id=data["id"],
            name=data["name"],
            status=data["status"],
            task_count=data.get("task_count", 0)
        )

@dataclass
class ProjectDTO:
    id: str
    name: str
    description: str
    root_path: str
    status: str
    task_count: int
    worker_count: int
    created_at: str
    updated_at: str

    def to_dict(self) -> Dict[str, Any]:
        return {
            "id": self.id,
            "name": self.name,
            "description": self.description,
            "root_path": self.root_path,
            "status": self.status,
            "task_count": self.task_count,
            "worker_count": self.worker_count,
            "created_at": self.created_at,
            "updated_at": self.updated_at
        }

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> ProjectDTO:
        return cls(
            id=data["id"],
            name=data["name"],
            description=data["description"],
            root_path=data["root_path"],
            status=data["status"],
            task_count=data.get("task_count", 0),
            worker_count=data.get("worker_count", 0),
            created_at=data["created_at"],
            updated_at=data["updated_at"]
        )

    @classmethod
    def from_domain(cls, project: Project, task_count: int = 0, worker_count: int = 0) -> ProjectDTO:
        return cls(
            id=project.id,
            name=project.name,
            description=project.description,
            root_path=project.root_path,
            status=project.status.value if hasattr(project.status, 'value') else str(project.status),
            task_count=task_count,
            worker_count=worker_count,
            created_at=project.created_at,
            updated_at=project.updated_at
        )

@dataclass
class ProjectCreateRequest:
    name: str
    root_path: str
    description: Optional[str] = None
    configuration: Optional[Dict[str, Any]] = None

    def to_dict(self) -> Dict[str, Any]:
        return {
            "name": self.name,
            "root_path": self.root_path,
            "description": self.description,
            "configuration": self.configuration
        }

    @classmethod
    def from_dict(cls, data: Dict[str, Any]) -> ProjectCreateRequest:
        return cls(
            name=data["name"],
            root_path=data["root_path"],
            description=data.get("description"),
            configuration=data.get("configuration")
        )

File: client/lib/models/project.dart
class Project {
  final String id;
  final String name;
  final String description;
  final String rootPath;
  final String status;
  final int taskCount;
  final int workerCount;
  final String createdAt;
  final String updatedAt;

  const Project({
    required this.id,
    required this.name,
    this.description = '',
    required this.rootPath,
    this.status = 'ACTIVE',
    this.taskCount = 0,
    this.workerCount = 0,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Project.fromJson(Map<String, dynamic> json) {
    return Project(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      description: json['description'] as String? ?? '',
      rootPath: json['root_path'] as String? ?? '',
      status: json['status'] as String? ?? 'ACTIVE',
      taskCount: json['task_count'] as int? ?? 0,
      workerCount: json['worker_count'] as int? ?? 0,
      createdAt: json['created_at'] as String? ?? '',
      updatedAt: json['updated_at'] as String? ?? '',
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'description': description,
    'root_path': rootPath,
    'status': status,
    'task_count': taskCount,
    'worker_count': workerCount,
    'created_at': createdAt,
    'updated_at': updatedAt,
  };
}
